### PR TITLE
Increase icon maximum size to 24x24 in skin generator

### PR DESCRIPTION
**Pull request by dvdmrtnz**
As mentioned in https://github.com/organicmaps/organicmaps/pull/7281#issuecomment-1926701446, this PR reduces all -m svg icons to fit inside a 18x18 box (which was already done by the skin generator script) and allows icons to be up to 24x24 in size.

### CODE REVIEW

**Comment by pastk** [comment]
Please correct me if I'm wrong.
Before:
- some svg icons were bigger than 18x18px
- the skin gen script reduced them to 18x18px
After:
- those icons are reduced to 18x18px in svg (so no bigger svg icons as of now unless we add some new e.g. traffic lights)
- they won't be reduced by the skin gen script anymore
- the skin gen script will allow icons up to 24x24px

So this PR shouldn't result in any visual change, is that right?

**Comment by euf** [comment]
@pastk You are absolutely right, and all icons are now reduced to 18×18 px in svg, except two of them:
- `geyser-l.svg` now renders 21×21 px instead of 18×18 px
- `speedcam-alert-l.svg` now renders 24×24 px instead of 18×18 px

The resulting visual change will be noticeable for these two icons as well as for future `traffic_lights`, `stop`, etc. 

Old gen script downsized all icons of them to fit into 18×18, which also led to some mistakes in icon sizes.
New gen script downsizes all icons to 24×24, still being foolproof, but allowing to utilise bigger 24×24 `-l` icons in addition to current 18×18 `-m` icons.

File: skin_generator/generator.cpp [hunk 1]
```diff
@@ -99,14 +99,14 @@ void SkinGenerator::ProcessSymbols(std::string const & svgDataDir,
           QSize size = svgSize * (symbolSize.width() / 18.0); // 18.0 is the size of the -m SVG files
 
           // Fitting symbol into symbolSize, saving aspect ratio.
-          if (size.width() > symbolSize.width())
+          if (size.width() > symbolSize.width() * 24.0 / 18.0)
```
Comment (pastk) [inline skin_generator/generator.cpp:102]: Could you please add a code comment explaining why there is a "* 24 / 18"?

**Review by pastk — APPROVED** [review]

File: skin_generator/generator.cpp (hunk 1 above)
Comment (dvdmrtnz) [reply skin_generator/generator.cpp:102]: Added comment

Comment (dvdmrtnz) [reply skin_generator/generator.cpp]: I have just refactored this to make it clearer what is happening here

File: skin_generator/generator.cpp [hunk 2]
```diff
@@ -17,6 +17,10 @@ namespace tools
 {
 namespace
 {
+
+static const double kLargeIconSize = 24.0;  // Size of the -l SVG icons
+static const double kMediumIconSize = 18.0; // size of the -m SVG icons
```
Comment (vng) [inline skin_generator/generator.cpp:22]: nit: constexpr

File: skin_generator/generator.cpp [hunk 3]
```diff
@@ -95,24 +99,25 @@ void SkinGenerator::ProcessSymbols(std::string const & svgDataDir,
...
-            size.setWidth(static_cast<int>(w));
-            size.setHeight(symbolSize.height());
+            auto const w = static_cast<float>(svgSize.width()) * kLargeIconSize / svgSize.height();
+            svgSize.setWidth(static_cast<int>(w));
+            svgSize.setHeight(kLargeIconSize);
           }
 
+          // Scale symbol to required size
+          QSize symbolSize = symbolSizes[j];  // Size of the symbol in the PNG skin
+          QSize size = svgSize * (symbolSize.width() / kMediumIconSize);
```
Comment (vng) [inline skin_generator/generator.cpp:119]: nit: No need ```QSize symbolSize = symbolSizes[j];``` here. Use ```symbolSizes[j]``` in place.

**Review by vng — COMMENTED** [review]
Am I right that PNG skins was regenerated because of deleted icon?
The actual icons' size on skin is not changed?

**Comment by dvdmrtnz** [comment]
> Am I right that PNG skins was regenerated because of deleted icon? The actual icons' size on skin is not changed?

Only `speedcam-alert-l.svg` changes size

File: skin_generator/generator.cpp (hunk 2 above)
Comment (dvdmrtnz) [reply skin_generator/generator.cpp]: Done

File: skin_generator/generator.cpp (hunk 3 above)
Comment (dvdmrtnz) [reply skin_generator/generator.cpp]: Done

**Review by vng — APPROVED** [review]

**Comment by euf** [comment]
Updated wiki page
https://github.com/organicmaps/organicmaps/wiki/Icons:-clear-vs.-night,-sizes

**Pull request merged**